model.py

```diff
@@ -8557,6 +8557,36 @@ def update_credentials(self, _db, access_token, credential):
         _db.commit()
 
 
+class AuthenticationService(object):
+    """An AuthenticationService contains configuration for a third-party
+    service that can authenticate people. There will be a subclass of
+    AuthenticationService for each type of person that needs to be
+    authenticated.
+    """
+
+    id = Column(Integer, primary_key=True)
+
+    name = Column(Unicode, unique=True, nullable=False, index=True)
+
+    provider = Column(Unicode, nullable=False, index=True)
+
+    # Supported values for the 'provider' field
+    GOOGLE_OAUTH = 'Google OAuth'
+
+    PROVIDERS = [GOOGLE_OAUTH]
+
+    # Additional provider-specific settings are stored as
+    # stringified JSON in the 'settings' field
+    settings = Column(Unicode)
```

Comment:
We could use a Postgres JSON column for this, but I think we would have to update sqlalchemy, and the string's probably fine since we don't have to load it that often.